### PR TITLE
Release v1.0.0

### PR DESCRIPTION
**Pull request by ronald-jaepel**
# Open ToDos:
- [ ] write release notes

# Release notes draft:

# v1.0.0

**CADET-Python** has long been used as a file-based interface for **CADET-Core**. With this release, we’re moving to v1.0.0, reflecting its maturity and expanded functionality. This version introduces a C-API for directly reading simulation results from CADET-Core, enhancing data processing efficiency. Future releases will aim to add more features and further refine the interface.

This release requires Python 3.9+.

### New Features

- **C-API Support**: Added functionality to directly interface with CADET-Core, allowing for more efficient result retrieval.
- **Autodetect CADET**: Enhanced user experience by automating the detection of CADET installation.
- **Extended Test Suite**: Broadened test coverage to improve stability and ensure reliability.
- **Enhanced Documentation**: Added comprehensive docstrings and type annotations for improved readability and usability.
- **Improved Logging**: Enhanced logging to provide better debugging and diagnostic information.
- **PEP8 Formatting**: Codebase updated to conform to PEP8, improving code quality and readability.

## Breaking changes

- **Updated `cadet.run()` Return**: `cadet.run()` now returns an instance of the `ReturnInformation` class, which includes:
  - `return_code: int`: Status code for execution.
  - `error_message: str`: Message describing any errors encountered.
  - `log: str`: Detailed log of the run. 

All users are encouraged to upgrade to **CADET-Python** v1.0.0 to take advantage of these improvements in performance, usability, and code quality.

### CODE REVIEW

**Review by schmoelder — APPROVED** [review]
LGTM

**Comment by schmoelder** [comment]
let's wait for @lieres' approval, though.

**Review by lieres — APPROVED** [review]

**Pull request merged**
